src/network/live_configuration/offline_mode.rs

```diff
@@ -0,0 +1,29 @@
+// (C) Copyright IBM Corp. 2024.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::AppConfigurationOffline;
+
+/// Defines the behaviour of the client while the connection to the server
+/// is lost. In all cases the client will keep trying to reconnect forever.
+#[derive(Debug)]
+pub enum OfflineMode {
+    /// Returns errors when requesting features or evaluating them
+    Fail,
+
+    /// Return features and values from the latests configuration available
+    Cache,
+
+    /// Use the provided configuration.
+    FallbackData(AppConfigurationOffline),
```

Comment:
I think using here another appconfig client is a good alternative. We don't need to decide if here we provide a `Configuration`, a `ConfigurationJSON` or a `PathBuf`; just use another client and that's all 🚀 

...I'm using `AppConfigurationOffline`, but we could use here `AppConfigurationClient` and accept anything that behaves like a `AppConfigurationClient`, wdyt?

Reply:
I actually think keeping it `AppConfigurationOffline` for now is fine and communicates the intention.
If we would accept abstract type here, the user would not immediately know how to use this.
We can change at any time without breaking change when a real need/feature-request arises.

Reply:
But I like the idea. It re-uses a concept and makes it generic :-)